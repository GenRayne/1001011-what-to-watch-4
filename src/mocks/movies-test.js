export const movies = [
  {
    id: `1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    picture: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: Math.floor(Math.random() * (2020 - 1990) + 1990),
    rating: Number((Math.random() * 10).toFixed(1)),
    ratingsNumber: Math.ceil(Math.random() * 500),
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus modi consequatur quia sit voluptate molestiae corrupti natus similique ab eaque dolorem hic, harum eveniet beatae porro nobis laudantium quibusdam eos? Tenetur, beatae. Velit inventore reiciendis voluptates doloribus incidunt, provident ducimus blanditiis quae voluptatum porro dolor optio illo iusto consequuntur ipsum!`,
  },
  {
    id: `2`,
    title: `Bohemian Rhapsody`,
    picture: `img/bohemian-rhapsody.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: Math.floor(Math.random() * (2020 - 1990) + 1990),
    rating: Number((Math.random() * 10).toFixed(1)),
    ratingsNumber: Math.ceil(Math.random() * 500),
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus modi consequatur quia sit voluptate molestiae corrupti natus similique ab eaque dolorem hic, harum eveniet beatae porro nobis laudantium quibusdam eos? Tenetur, beatae. Velit inventore reiciendis voluptates doloribus incidunt, provident ducimus blanditiis quae voluptatum porro dolor optio illo iusto consequuntur ipsum!`,
  },
  {
    id: `3`,
    title: `Macbeth`,
    picture: `img/macbeth.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: Math.floor(Math.random() * (2020 - 1990) + 1990),
    rating: Number((Math.random() * 10).toFixed(1)),
    ratingsNumber: Math.ceil(Math.random() * 500),
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus modi consequatur quia sit voluptate molestiae corrupti natus similique ab eaque dolorem hic, harum eveniet beatae porro nobis laudantium quibusdam eos? Tenetur, beatae. Velit inventore reiciendis voluptates doloribus incidunt, provident ducimus blanditiis quae voluptatum porro dolor optio illo iusto consequuntur ipsum!`,
  },
  {
    id: `4`,
    title: `Aviator`,
    picture: `img/aviator.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: Math.floor(Math.random() * (2020 - 1990) + 1990),
    rating: Number((Math.random() * 10).toFixed(1)),
    ratingsNumber: Math.ceil(Math.random() * 500),
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus modi consequatur quia sit voluptate molestiae corrupti natus similique ab eaque dolorem hic, harum eveniet beatae porro nobis laudantium quibusdam eos? Tenetur, beatae. Velit inventore reiciendis voluptates doloribus incidunt, provident ducimus blanditiis quae voluptatum porro dolor optio illo iusto consequuntur ipsum!`,
  },
  {
    id: `5`,
    title: `We need to talk about Kevin`,
    picture: `img/we-need-to-talk-about-kevin.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: Math.floor(Math.random() * (2020 - 1990) + 1990),
    rating: Number((Math.random() * 10).toFixed(1)),
    ratingsNumber: Math.ceil(Math.random() * 500),
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus modi consequatur quia sit voluptate molestiae corrupti natus similique ab eaque dolorem hic, harum eveniet beatae porro nobis laudantium quibusdam eos? Tenetur, beatae. Velit inventore reiciendis voluptates doloribus incidunt, provident ducimus blanditiis quae voluptatum porro dolor optio illo iusto consequuntur ipsum!`,
  },
  {
    id: `6`,
    title: `What We Do in the Shadows`,
    picture: `img/what-we-do-in-the-shadows.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: Math.floor(Math.random() * (2020 - 1990) + 1990),
    rating: Number((Math.random() * 10).toFixed(1)),
    ratingsNumber: Math.ceil(Math.random() * 500),
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus modi consequatur quia sit voluptate molestiae corrupti natus similique ab eaque dolorem hic, harum eveniet beatae porro nobis laudantium quibusdam eos? Tenetur, beatae. Velit inventore reiciendis voluptates doloribus incidunt, provident ducimus blanditiis quae voluptatum porro dolor optio illo iusto consequuntur ipsum!`,
  },
  {
    id: `7`,
    title: `Revenant`,
    picture: `img/revenant.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: Math.floor(Math.random() * (2020 - 1990) + 1990),
    rating: Number((Math.random() * 10).toFixed(1)),
    ratingsNumber: Math.ceil(Math.random() * 500),
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus modi consequatur quia sit voluptate molestiae corrupti natus similique ab eaque dolorem hic, harum eveniet beatae porro nobis laudantium quibusdam eos? Tenetur, beatae. Velit inventore reiciendis voluptates doloribus incidunt, provident ducimus blanditiis quae voluptatum porro dolor optio illo iusto consequuntur ipsum!`,
  },
  {
    id: `8`,
    title: `Johnny English`,
    picture: `img/johnny-english.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: Math.floor(Math.random() * (2020 - 1990) + 1990),
    rating: Number((Math.random() * 10).toFixed(1)),
    ratingsNumber: Math.ceil(Math.random() * 500),
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus modi consequatur quia sit voluptate molestiae corrupti natus similique ab eaque dolorem hic, harum eveniet beatae porro nobis laudantium quibusdam eos? Tenetur, beatae. Velit inventore reiciendis voluptates doloribus incidunt, provident ducimus blanditiis quae voluptatum porro dolor optio illo iusto consequuntur ipsum!`,
  },
];

export const someMovie = {
  id: `1`,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  picture: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `Adventures`,
  releaseYear: 2018,
};
