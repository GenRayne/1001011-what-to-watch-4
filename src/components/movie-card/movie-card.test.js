import React from 'react';
import renderer from 'react-test-renderer';
import MovieCard from './movie-card.jsx';
import {someMovie} from '../../mocks/movies-test.js';

describe(`render MovieCard`, () => {
  it(`renders MovieCard`, () => {
    const handleTitleClick = jest.fn();
    const handleCardMouseEnter = jest.fn();
    const handleCardMouseLeave = jest.fn();

    const tree = renderer
      .create(
          <MovieCard
            movie={someMovie}
            onTitleClick={handleTitleClick}
            onCardMouseEnter={handleCardMouseEnter}
            onCardMouseLeave={handleCardMouseLeave}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
