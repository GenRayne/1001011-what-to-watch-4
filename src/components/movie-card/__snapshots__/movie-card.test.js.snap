// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MovieCard renders MovieCard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[MockFunction]}
>
  <a
    href="#"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Poster for: Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      onClick={[Function]}
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
